CRM/Core/BAO/Setting.php

```diff
@@ -208,6 +211,9 @@ public static function setItems(&$params, $domains = NULL) {
     $fieldsToSet = self::validateSettingsInput($params, $fields);
 
     foreach ($fieldsToSet as $settingField => &$settingValue) {
+      if (self::isEnvironmentSet($settingField)) {
```

Comment:
Once again, I'm not sure why this field is handled differently. Shouldn't this test be put into self::validateSetting?